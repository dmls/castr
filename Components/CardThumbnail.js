import React from 'react';
import { View, Text, Image, TouchableOpacity } from 'react-native';
import { styles } from '../assets/styles/Styles';

const CardThumbnail = ({ index, onPress, data, actions, fullWidth = false }) => {
  const CardTitle = (
    <View style={{flex: 2, flexDirection: 'column', justifyContent: 'space-between'}}>
      <View>
        <Text style={styles.h2}>{data.name}</Text>
      </View>

      {!fullWidth && (<View>{actions}</View>)}
    </View>
  );

  const CardImage = data.image ? (
    <View style={{ flex: 1, ...(fullWidth ? {} : { height: 100 }) }}>
      <Image source={{uri: data.image}} style={[styles.imageFullWidth]} />
    </View>
  ) : (<></>);
  
  return (
    <TouchableOpacity key={index} onPress={onPress} style={styles.section}>

      {fullWidth && (
        <>
          <View style={styles.sectionRow}>
            {CardTitle}
          </View>

          {CardImage}
        </>
      )}

      {!fullWidth && (
        <View style={styles.sectionRow}>
          {CardTitle}
          {CardImage}
        </View>
      )}
    </TouchableOpacity>  
  );
}

export default CardThumbnail;
