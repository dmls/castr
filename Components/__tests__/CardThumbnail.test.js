import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

import CardThumbnail from '../CardThumbnail';

const index = 0;
const onPress = jest.fn();
const data = {
  name: 'data name',
  image: 'file://image-path-here.jpg',
};
const actions = (
  <>
    <Text>Sample action 1</Text>
    <Text>Sample action 2</Text>
  </>
);

describe('CardThumbnail', () => {
  test('renders correctly with default settings', () => {
    const component = renderer.create(
      <CardThumbnail
        index={index}
        onPress={onPress}
        data={data}
        actions={actions}
      />
    );
    const tree = component.toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with fullWidth = false', () => {
    const component = renderer.create(
      <CardThumbnail
        index={index}
        onPress={onPress}
        data={data}
        actions={actions}
        fullWidth={false}
      />
    );
    const tree = component.toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with fullWidth = true', () => {
    const component = renderer.create(
      <CardThumbnail
        index={index}
        onPress={onPress}
        data={data}
        actions={actions}
        fullWidth={true}
      />
    );
    const tree = component.toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});
