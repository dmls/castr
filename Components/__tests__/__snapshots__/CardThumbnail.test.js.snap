// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardThumbnail renders correctly with default settings 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignContent": "center",
      "backgroundColor": "#fff",
      "flexDirection": "column",
      "margin": 4,
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignContent": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": 4,
        "marginVertical": 10,
        "paddingHorizontal": 10,
      }
    }
  >
    <View
      style={
        {
          "flex": 2,
          "flexDirection": "column",
          "justifyContent": "space-between",
        }
      }
    >
      <View>
        <Text
          style={
            {
              "fontSize": 20,
            }
          }
        >
          data name
        </Text>
      </View>
      <View>
        <Text>
          Sample action 1
        </Text>
        <Text>
          Sample action 2
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flex": 1,
          "height": 100,
        }
      }
    >
      <Image
        source={
          {
            "uri": "file://image-path-here.jpg",
          }
        }
        style={
          [
            {
              "aspectRatio": 1.3333333333333333,
              "flex": 1,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`CardThumbnail renders correctly with fullWidth = false 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignContent": "center",
      "backgroundColor": "#fff",
      "flexDirection": "column",
      "margin": 4,
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignContent": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": 4,
        "marginVertical": 10,
        "paddingHorizontal": 10,
      }
    }
  >
    <View
      style={
        {
          "flex": 2,
          "flexDirection": "column",
          "justifyContent": "space-between",
        }
      }
    >
      <View>
        <Text
          style={
            {
              "fontSize": 20,
            }
          }
        >
          data name
        </Text>
      </View>
      <View>
        <Text>
          Sample action 1
        </Text>
        <Text>
          Sample action 2
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flex": 1,
          "height": 100,
        }
      }
    >
      <Image
        source={
          {
            "uri": "file://image-path-here.jpg",
          }
        }
        style={
          [
            {
              "aspectRatio": 1.3333333333333333,
              "flex": 1,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`CardThumbnail renders correctly with fullWidth = true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignContent": "center",
      "backgroundColor": "#fff",
      "flexDirection": "column",
      "margin": 4,
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignContent": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": 4,
        "marginVertical": 10,
        "paddingHorizontal": 10,
      }
    }
  >
    <View
      style={
        {
          "flex": 2,
          "flexDirection": "column",
          "justifyContent": "space-between",
        }
      }
    >
      <View>
        <Text
          style={
            {
              "fontSize": 20,
            }
          }
        >
          data name
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Image
      source={
        {
          "uri": "file://image-path-here.jpg",
        }
      }
      style={
        [
          {
            "aspectRatio": 1.3333333333333333,
            "flex": 1,
          },
        ]
      }
    />
  </View>
</View>
`;
